app/channels.go

```diff
@@ -78,8 +78,11 @@ type Channels struct {
 	postReminderMut  sync.Mutex
 	postReminderTask *model.ScheduledTask
 
-	collectionTypes map[string]string   // pluginId -> collectionType
-	topicTypes      map[string][]string // collectionType -> topicType array
+	// collectionTypes maps collection types array to the registering plugin
+	collectionTypes map[string][]string
+	// topicTypes maps topic types array to collection types
+	topicTypes    map[string][]string
+	collectionMut sync.Mutex
```

Comment:
It looks like this lock covers both the `collectionTypes` and `topicTypes` writes -- should we call it `collectionAndTopicTypesMut` for clarity?